src/client.rs

```diff
@@ -0,0 +1,162 @@
+use std::{env, fmt};
+use std::sync::Arc;
+
+#[cfg(not(feature = "blocking"))]
+use reqwest::{Client as ReqwestClient, RequestBuilder};
+use reqwest::{Method, Url};
+#[cfg(feature = "blocking")]
+use reqwest::blocking::{Client as ReqwestClient, RequestBuilder};
+use reqwest::header::USER_AGENT;
+
+use crate::services::{ApiKeys, Audiences, Contacts, Domains, Emails};
+
+/// A minimal [Resend](https://resend.com) client.
+// TODO: Arc<ClientInner> + impl Deref?
+#[must_use]
+#[derive(Clone)]
+pub struct Client {
+    /// `Resend` APIs for `METHOD /emails` endpoints.
+    pub emails: Emails,
+    /// `Resend` APIs for `METHOD /api-keys` endpoints.
+    pub api_keys: ApiKeys,
+    /// `Resend` APIs for `METHOD /audiences` endpoints.
+    pub audiences: Audiences,
+    /// `Resend` APIs for `METHOD /audiences/:id/contacts` endpoints.
+    pub contacts: Contacts,
+    /// `Resend` APIs for `METHOD /domains` endpoints.
+    pub domains: Domains,
+}
+
+#[derive(Clone)]
+pub struct Config {
+    user_agent: String,
+    api_key: String,
+    pub(crate) base_url: Url,
+    pub(crate) client: ReqwestClient,
+}
+
+impl Config {
+    /// Constructs a new [`RequestBuilder`] with parameters.
+    pub fn build(&self, method: Method, path: &str) -> RequestBuilder {
+        let path = self
+            .base_url
+            .join(path)
+            .expect("should a valid API endpoint");
+
+        self.client
+            .request(method, path)
+            .bearer_auth(self.api_key.as_str())
+            .header(USER_AGENT, self.user_agent.as_str())
+    }
+}
+
+impl fmt::Debug for Config {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // Don't output API key.
+        f.debug_struct("Client")
+            .field("api_key", &"re_*********")
+            .field("user_agent", &self.user_agent.as_str())
+            .field("base_url", &self.base_url.as_str())
+            // .field("client", &self.client)
+            .finish()
+    }
+}
+
+impl Client {
+    /// Creates a new [`Resend`] client.
+    ///
+    /// ### Panics
+    ///
+    /// Panics if the environment variable `RESEND_BASE_URL` is present but is not a valid URL.
+    ///
+    /// [`Resend`]: https://resend.com
+    pub fn new(api_key: &str) -> Self {
+        Self::with_client(api_key, ReqwestClient::default())
+    }
+
+    /// Creates a new [`Resend`] client with a provided [`reqwest::Client`].
+    ///
+    /// ### Panics
+    ///
+    /// Panics if the environment variable `RESEND_BASE_URL` is present but is not a valid URL.
+    ///
+    /// [`Resend`]: https://resend.com
+    /// [`reqwest::Client`]: ReqwestClient
```

Comment:
This line seems to actually break the doc comment, removing it fixes the link, not sure if you meant to have these the other way around maybe

Reply:
What line, exactly? Both links seem to work for me.
Here is what I see in IDEA and cargo doc:
https://imgur.com/a/HGCMkO1

Reply:
Ok today I learned vscode does this wrong lol, if I built the docs, it works as it is now and breaks if you remove the last line. No clue why vscode apparently does its own thing here.